workspaces/ui-v2/src/store/documentationEdit/slice.ts

```diff
@@ -68,6 +76,45 @@ const documentationEditSlice = createSlice({
         };
       }
     },
+    // TODO FLEB connect this up to the UI
+    removeContribution: (
+      state,
+      action: PayloadAction<{
+        id: string;
+        contributionKey: string;
+      }>
+    ) => {
+      const { id, contributionKey } = action.payload;
+      if (state.contributions[id]?.[contributionKey]) {
+        delete state.contributions[id][contributionKey];
+      }
+    },
+    // TODO FLEB connect this up to the UI
+    addFieldEdit: (
+      state,
+      action: PayloadAction<{
+        fieldId: string;
+        options: {
+          isOptional: boolean;
+          isNullable: boolean;
+        };
+      }>
+    ) => {
+      const { fieldId, options } = action.payload;
+      state.fields.edited[fieldId] = options;
```

Comment:
Is there any benefit to checking whether the field has been removed before accepting edits, or is that something we don't want to constrain here?

Reply:
This is more of an upsert, rather than an add. We could presumably set isOptional: true and then also isNullable true. It's only if we have the default values, we want to remove this in the edited state (since we have no commands to generate)